app/services/hyrax/collections/permissions_service.rb

```diff
@@ -1,77 +1,118 @@
 module Hyrax
   module Collections
     class PermissionsService
+      # @api private
+      #
+      # IDs of collections, including admin sets, a user can access based on participant roles.
+      #
+      # @param user [User] user
+      # @param access [Array<String>] one or more types of access (e.g. Hyrax::PermissionTemplateAccess::MANAGE, Hyrax::PermissionTemplateAccess::DEPOSIT, Hyrax::PermissionTemplateAccess::VIEW)
+      # @return [Array<String>] IDs of collections for which the user has specified roles
+      def self.collection_ids_for_user(user:, access:)
+        if user.ability.admin?
+          PermissionTemplate.all.where(source_type: 'collection').pluck('DISTINCT source_id')
+        else
+          PermissionTemplateAccess.joins(:permission_template)
+              .where(agent_type: 'user',
+                     agent_id: user.user_key,
+                     access: access)
+              .or(
+                  PermissionTemplateAccess.joins(:permission_template)
+                      .where(agent_type: 'group',
+                             agent_id: user.groups,
+                             access: access)
+              ).pluck('DISTINCT source_id')
+        end
+      end
+      private_class_method :collection_ids_for_user
+
       # @api public
       #
-      # Get a list of users who should be added as user editors for a collection
+      # IDs of collections, including admin sets, for which the user is assigned view access
       #
-      # @param collection [Hyrax::Collection] the collection for which permissions are being set
-      # @return [Array<String>] array of user identifiers (typically emails) for users who can edit this collection
-      def self.user_edit_grants_for_collection(collection: nil)
-        return [] unless collection
```

Comment:
NOTE: I did not find a need for the previously anticipated service methods in the form `user_edit_grants_for_collection`.  This is primarily the case because there is a simple way to perform this query directly from the permission_template.

```
edit_users += template.agent_ids_for(agent_type: 'user', access: 'manage')
edit_groups += template.agent_ids_for(agent_type: 'group', access: 'manage')
read_users += template.agent_ids_for(agent_type: 'user', access: 'view')
read_groups += template.agent_ids_for(agent_type: 'group', access: 'view')
```